src/main/java/emissary/output/DropOffPlace.java

```diff
@@ -246,7 +246,7 @@ public List<IBaseDataObject> agentProcessHeavyDuty(final List<IBaseDataObject> p
                 objectMetricsLog.info(appendEntries(outputObjectMetrics(tld, objectMetricsFields)), "Finished DropOff");
             }
 
-            if (outputCompletionPayloadSize) {
+            if (outputCompletionPayloadSize && tld.data() != null) {
```

Comment:
Updating line 253 tld.data().length to tld.dataLength() and removing the null check should also address the NPE, with the benefit outputting the log message
```suggestion
            if (outputCompletionPayloadSize) {
```

Reply:
I think this still (in a test scenario) can throw an NPE due based on how the `dataLength()` is implemented. The use of this method would be preferred however, so I think both should be done.